### DISCUSSION TITLE
max_request_bytes on GATEWAY or SIDECAR_INBOUND

### DISCUSSION THREAD

**Discussion by jostockley** [istio/istio#48478] [Q&A]
Hi,
We want to set the maximum request body size for incoming requests from browser applications.
We want different maximum limits for different workloads, for example 5M for app1, but 256M for app2.
So we need two EnvoyFilters with workloadSelector for app1 and app2 respectively.

Do we apply the EnvoyFilters to the GATEWAY context only, the SIDECAR INBOUND context only or ANY?

What is the default maximum request body size and is the default the same for GATEWAY and SIDECAR_INBOUND?

**Comment by rootsongjc** [comment]
The configuration in the `GATEWAY` context applies to the entire gateway, affecting all traffic passing through that gateway rather than the traffic of specific services.

If you need to enforce the limits right before the requests hit your applications, you should use `SIDECAR_INBOUND`.

About the maximum request body size, see this issue: https://github.com/envoyproxy/envoy/issues/7650
